### ISSUE TITLE
Building Guix Package With CCL

### ISSUE THREAD

**Issue by hendursaga** [atlas-engineer/nyxt#2592]
Not so much a bug report as a question: how do I build the Guix package to use CCL instead of SBCL?

There's a [comment](https://github.com/atlas-engineer/nyxt//blob/aef5118a7c30c48583b464e2e0c27d4b0060999b/build-scripts/nyxt.scm#L121-L122) that mentions what I'm trying to do. I'm presuming replacing `sbcl` as a native-input is one step, but are there any others? [Another comment](https://github.com/atlas-engineer/nyxt//blob/aef5118a7c30c48583b464e2e0c27d4b0060999b/build-scripts/nyxt.scm#L86) mentions that stripping breaks SBCL binaries, but is this also true for CCL?

**Comment by Ambrevar** [comment]
To clarify: do you just want to use `build-scripts/nyxt.scm` or do you actually want to _produce_ a Guix package?

For the former, it's easy, just load the scm in a `guix shell` with CCL and load the system.

For the latter, unfortunately CCL breaks with cl-cffi-gtk (or cl-webkit2?).  See https://github.com/atlas-engineer/nyxt/issues/1057.

The way forward here, I believe, would be to get rid of cl-cffi-gtk and use only cl-gobject-introspection.  It's a relatively easy task.

**Comment by Ambrevar** [comment]
> [Another comment](https://github.com/atlas-engineer/nyxt//blob/aef5118a7c30c48583b464e2e0c27d4b0060999b/build-scripts/nyxt.scm#L86) mentions that stripping breaks SBCL binaries, but is this also true for CCL?

I've never tried :)

**Comment by hendursaga** [comment]
Yeah, I was meaning more of the latter, although I suppose the former could suffice in some cases. And yes, I do think using only `cl-gobject-introspection` would be the way forward!
